src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/validation/CredentialsValidator.java

```diff
@@ -53,39 +54,45 @@ public boolean validate() {
-                && regex != null
-                && !regex.isEmpty()
                 && this.content != null
                 && this.content.length() > 1) {
             try {
                 final Map<String, Object> contentAsMap = XContentHelper.convertToMap(this.content, false, XContentType.JSON).v2();
                 if (contentAsMap != null && contentAsMap.containsKey("password")) {
                     final String password = (String) contentAsMap.get("password");
 
-                    // Password can be null/empty for an existing user. Regex will validate password if present
-                    if (password != null && !password.isEmpty() && !regex.isEmpty() && !Pattern.compile("^"+regex+"$").matcher(password).matches()) {
-                        if(log.isDebugEnabled()) {
-                            log.debug("Regex does not match password");
-                        }
+                    // Password is not allowed to be empty if present.
+                    if (password != null && password.isEmpty()) {
                         this.errorType = ErrorType.INVALID_PASSWORD;
                         return false;
                     }
 
-                    final String username = Utils.coalesce(request.param("name"), hasParams() ? (String) param[0] : null);
+                    if (!Strings.isNullOrEmpty(regex)) {
+                        // Password can be null for an existing user. Regex will validate password if present
+                        if (password != null && !Pattern.compile("^"+regex+"$").matcher(password).matches()) {
+                            if(log.isDebugEnabled()) {
+                                log.debug("Regex does not match password");
+                            }
+                            this.errorType = ErrorType.INVALID_PASSWORD;
+                            return false;
+                        }
+
+                        final String username = Utils.coalesce(request.param("name"), hasParams() ? (String) param[0] : null);
 
-                    if (username == null || username.isEmpty()) {
-                        if (log.isDebugEnabled()) {
-                            log.debug("Unable to validate username because no user is given");
+                        if (username == null || username.isEmpty()) {
+                            if (log.isDebugEnabled()) {
+                                log.debug("Unable to validate username because no user is given");
+                            }
+                            return false;
                         }
-                        return false;
-                    }
 
-                    if (username.toLowerCase().equals(password.toLowerCase())) {
-                        if (log.isDebugEnabled()) {
-                            log.debug("Username must not match password");
+                        if (username.toLowerCase().equals(password.toLowerCase())) {
```

Comment:
Can password be null here? 
Maybe check for password is not null directly in the contentAsMap as these conditions are based on non-null password entry validation

Reply:
Also not dependent on regex but it is inside `Strings.isNullOrEmpty(regex)` block? I understand the current behavior validates password against username only if regex is defined. Just looking for improvements here. Any thoughts?

Reply:
Yes, I thought about that too. In theory, password has nothing to do with regex. I feel like there're some points/use cases to validate password against username no matter regex is null/empty or not and refactoring could be needed here. It's worth to discuss with the team and to get more inputs.

Reply:
Can we address in this PR itself as change is happening around it

Reply:
I'll create another PR to make this change given that this PR has been sitting there for more than 3 weeks and it has met the goal. Let's stick with the original scope of PRs.